test/functional/feature_assumeutxo.py

```diff
@@ -218,6 +222,27 @@ def test_snapshot_block_invalidated(self, dump_output_path):
             assert_raises_rpc_error(-32603, msg, node.loadtxoutset, dump_output_path)
             node.reconsiderblock(block_hash)
 
+    def test_snapshot_not_on_most_work_chain(self, dump_output_path):
+        self.log.info("Test bitcoind should fail when the node knows the headers of another chain with more work.")
+        node0 = self.nodes[0]
+        node1 = self.nodes[1]
+        # create two blocks that form a longer chain
+        parent = node0.getblockhash(SNAPSHOT_BASE_HEIGHT - 1)
+        block_time = node0.getblock(node0.getbestblockhash())['time'] + 1
+        fork1 = create_block(int(parent, 16), create_coinbase(SNAPSHOT_BASE_HEIGHT - 1), block_time)
+        fork1.solve()
+        fork2 = create_block(fork1.sha256, create_coinbase(SNAPSHOT_BASE_HEIGHT), block_time + 1)
+        fork2.solve()
+        node1.submitheader(fork1.serialize().hex())
+        node1.submitheader(fork2.serialize().hex())
+        msg = "A forked headers-chain with more work than the chain with the snapshot base block header exists. Please proceed to sync without AssumeUtxo."
+        assert_raises_rpc_error(-32603, msg, node1.loadtxoutset, dump_output_path)
+        # Cleanup: submit two more headers of the snapshot chain, so that loading the snapshot in future subtests succeeds
+        main1 = node0.getblock(node0.getblockhash(SNAPSHOT_BASE_HEIGHT + 1), 0)
```

Comment:
nit: making the variables a bit more detailed could make the test a bit easier to parse

`parent` = > `parent_block_hash`
`fork1` => `fork_block1`
`main1` = `main_block1`

Reply:
took those